src/git.rs

```diff
@@ -79,40 +115,181 @@ impl Git for LiveGit {
             None => Err(Box::new(git2::Error::from_str("No branch name found")))
         }
     }
-}
 
-#[cfg(test)]
-mod test {
-    use super::*;
-    
-    pub struct MockGit;
-    impl Git for MockGit {
-        fn new() -> Result<Self, Box<dyn Error>> {
-            Ok(MockGit)
+    fn log(&self) -> Result<(), Box<dyn Error>> {
+        let mut revwalk = self.repo.revwalk()?;
+        revwalk.push_head()?;
+        for id in revwalk {
+            let id = id?;
+            let commit = self.repo.find_commit(id)?;
+            Self::debug_commit(&commit);
+        }
+        Ok(())
+    }
+
+    fn find_commit(&self, commit_id: Oid) -> Result<Commit<'_>, Box<dyn Error>> {
+        let new_commit = self.repo.find_commit(commit_id)?;
+        Ok(new_commit)                
+    }
+
+    fn find_commit_fingerprints(&self, team_fingerprint_file: &str, commits: &Vec<Commit<'_>>) -> Result<HashSet<String>, Box<dyn Error>> {
+
+        let team_fingerprints = Fingerprint::read_fingerprints::<LiveGit>(self, team_fingerprint_file)?;
+
+        let mut commit_fingerprints = HashSet::new();
+
+        for commit in commits.iter() {
+            let author = commit.author();
+            let commit_email = match author.email() {
+                Some(e) => e,
+                None => return Err(Box::new(CapnError::new(format!("Email on commit not found"))))
+            };
+            let fingerprint = team_fingerprints.iter().find(|f| f.email == commit_email);
```

Comment:
This could get very slow in a large branch.  See previous comment about making team_fingerprints a hashmap.